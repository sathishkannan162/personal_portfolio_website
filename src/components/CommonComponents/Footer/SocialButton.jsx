import React from 'react';
import { chakra, VisuallyHidden, useColorModeValue } from '@chakra-ui/react';

const SocialButton = ({ children, label, href, ...props }) => {
  return (
    <chakra.button
      bg={useColorModeValue('blackAlpha.100', 'whiteAlpha.100')}
      rounded={'full'}
      w={8}
      h={8}
      cursor={'pointer'}
      as={'a'}
      href={href}
      display={'inline-flex'}
      alignItems={'center'}
      justifyContent={'center'}
      transition={'background 0.3s ease'}
      _hover={{
        bg: useColorModeValue('blackAlpha.200', 'whiteAlpha.200'),
      }}
      {...props}
    >
      <VisuallyHidden>{label}</VisuallyHidden>
      {children}
    </chakra.button>
  );
};

export default SocialButton;

