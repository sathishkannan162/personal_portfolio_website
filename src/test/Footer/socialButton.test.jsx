import React from 'react';
import { render, screen } from '../test-utils'
import SocialButton from '../../components/CommonComponents/Footer/SocialButton'

describe('SocialButton', () => {
  it('renders the button with correct label and href', () => {
    const label = 'F';
    const href = 'https://www.facebook.com';

    render(
      <SocialButton label={label} href={href}>
        <span>Facebook Icon</span>
      </SocialButton>
    );

    const button = screen.getByRole('link');
    expect(button).toHaveTextContent(label)
    expect(button).toBeInTheDocument();
    expect(button).toHaveAttribute('href', href);

    const icon = screen.getByText('Facebook Icon');
    expect(icon).toBeInTheDocument();
  });
  it('captures and compares component snapshot', () => {
    const { asFragment } = render(<SocialButton />);
    expect(asFragment()).toMatchSnapshot();
  });
});
