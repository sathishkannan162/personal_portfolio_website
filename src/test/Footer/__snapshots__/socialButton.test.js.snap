// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialButton captures and compares component snapshot 1`] = `
<DocumentFragment>
  <a
    class="css-hyb7bs"
  >
    <span
      class="css-idkz9h"
    />
  </a>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
